packages/calcite-components/src/components/popover/popover.tsx

```diff
@@ -288,16 +286,19 @@ export class Popover
   //
   // --------------------------------------------------------------------------
 
-  connectedCallback(): void {
+  async connectedCallback(): Promise<void> {
     this.setFilteredPlacements();
     connectLocalized(this);
     connectMessages(this);
-    this.setUpReferenceElement(this.hasLoaded);
+
+    await componentOnReady(this.el);
+    this.setUpReferenceElement();
     connectFocusTrap(this);
+    connectFloatingUI(this, this.effectiveReferenceElement, this.el);
```

Comment:
`setUpReferenceElement` already calls `connectFloatingUI` so this may not be needed here.